worldedit-core/src/main/java/com/fastasyncworldedit/core/util/TextureUtil.java

```diff
@@ -555,6 +600,85 @@ private static long getDistance(BufferedImage image, int c1) {
+     * @since TODO
+     */
+    private static VersionMetadata getLatestVersion() throws IOException {
+        try (BufferedInputStream in = new BufferedInputStream(new URL(VERSION_MANIFEST).openStream());
+             BufferedReader reader = new BufferedReader(new InputStreamReader(in))) {
+            final JsonElement element = JsonParser.parseReader(reader);
+            for (final JsonElement versions : element.getAsJsonObject().getAsJsonArray("versions")) {
+                JsonObject version = versions.getAsJsonObject();
+                if (!version.get("type").getAsString().equals("release")) {
+                    continue;
+                }
+                final String clientJsonUrl = version.get("url").getAsString();
+                final String id = version.get("id").getAsString();
+                return new VersionMetadata(id, clientJsonUrl);
+            }
+        }
+        throw new IOException("Failed to get latest version metadata");
+    }
+
+    /**
+     * Retrieves the url to the client.jar based on the previously retrieved {@link VersionMetadata}
+     *
+     * @param metadata The version metadata containing the url to the client.jar
+     * @return The full url to the client.jar including the expected file hash for validation purposes
+     * @throws IOException If any http / i/o operation fails.
+     * @since TODO
+     */
+    private static HashedResource getLatestClientJarUrl(VersionMetadata metadata) throws IOException {
+        try (BufferedInputStream in = new BufferedInputStream(new URL(metadata.url()).openStream());
+             BufferedReader reader = new BufferedReader(new InputStreamReader(in))) {
+            final JsonObject object = JsonParser.parseReader(reader).getAsJsonObject();
+            final JsonObject client = object.getAsJsonObject("downloads").getAsJsonObject("client");
+            return new HashedResource(client.get("url").getAsString(), client.get("sha1").getAsString());
+        }
+    }
+
+    /**
+     * Calculates the sha-1 hash based on the content of the provided file.
+     *
+     * @param file The file to generate the sha-1 hash for
+     * @return The hash of the file contents
+     * @throws NoSuchAlgorithmException If the SHA-1 algorithm could not be resolved
+     * @throws IOException              If any I/O operation failed
+     * @since TODO
+     */
+    private static String calculateSha1(File file) throws NoSuchAlgorithmException, IOException {
+        MessageDigest digest = MessageDigest.getInstance("SHA-1");
+        try (final BufferedInputStream stream = new BufferedInputStream(new FileInputStream(file));
+             final DigestInputStream digestInputStream = new DigestInputStream(stream, digest)) {
+            // noinspection StatementWithEmptyBody - update digest with bytes from passed input stream
+            while (digestInputStream.read() != -1) {
+            }
+            return toHexString(digest.digest());
+        }
```

Comment:
```suggestion
        return toHexString(digest.digest(Files.readAllBytes(file.toPath())));
```
In general, using Path instead of File where possible would be great.

Reply:
This still uses the DigestInputStream, but from a second look, reading the bytes lazy instead all at once might be better - but I think we can also use DigestOutputStream and use transferTo here too? This would at least avoid the while loop.